ios/brave-ios/Sources/BraveStore/Subscription/SDK/AppStoreSDK.swift

```diff
@@ -339,6 +339,22 @@ public class AppStoreSDK: ObservableObject {
     (try? await subscription(for: product)?.subscription?.status) ?? []
   }
 
+  /// Retrieves a product's renewable subscription trial status
+  /// - Parameter product: The product whose subscription trial status to retrieve
+  /// - Returns: The renewable subscription's trial status
+  @MainActor
```

Comment:
Why are all these function `@MainActor` isolated?  None of it seems to require it